src/server/command_registry.cc

```diff
@@ -35,11 +35,48 @@ using absl::StrAppend;
 using absl::StrCat;
 using absl::StrSplit;
 
-CommandId::CommandId(const char* name, uint32_t mask, int8_t arity, int8_t first_key,
-                     int8_t last_key, uint32_t acl_categories)
-    : facade::CommandId(name, mask, arity, first_key, last_key, acl_categories) {
+namespace {
+
+uint32_t ImplicitCategories(uint32_t mask) {
   if (mask & CO::ADMIN)
-    opt_mask_ |= CO::NOSCRIPT;
+    mask |= CO::NOSCRIPT;
+  return mask;
+}
+
+uint32_t ImplicitAclCategories(uint32_t mask) {
+  mask = ImplicitCategories(mask);
+  uint32_t out = 0;
+
+  if (mask & CO::WRITE)
+    out |= acl::WRITE;
+
+  if ((mask & CO::READONLY) && ((mask & CO::NOSCRIPT) == 0))
+    out |= acl::READ;
+
+  if (mask & CO::ADMIN)
+    out |= acl::ADMIN | acl::DANGEROUS;
+
+  // todo pubsub
+
+  if (mask & CO::FAST)
+    out |= acl::FAST;
+
+  if (mask & CO::BLOCKING)
+    out |= acl::BLOCKING;
+
+  if ((out & acl::FAST) == 0)
```

Comment:
A counter example to this is `HGETALL` which we mark as:

```
<< CI{"HGETALL", CO::FAST | CO::READONLY, 2, 1, 1, acl::kHGetAll}.HFUNC(HGetAll)
```

which according to the rules above would mark it as `FAST` instead of `SLOW`. 

My concern here is that we have to manually check all of these commands and the `CO::` variants actually do map to the rules implemented in `ImplicitAclCategories` (and *if* the above mapping has no mistakes as well)

Another concern I have is that if we have bugs in our `CO::` variants the `Implicit` rules describe above will insert a `bugs` in our ACL rules.

Reply:
>  Another concern I have is that if we have bugs 

You can apply that concern to any code, I've added an option to return acl categories from COMMAND, so we can compare before and after and see the differences

[before.txt](https://github.com/user-attachments/files/16443724/before.txt)


> A counter example to this is HGETALL which we mark as:

It's not a counter example, we just remove CO::FAST. Again, the CO tags don't mean anything now

Reply:
Nice, in fact a `diff` on this will also hinter any errors. I like that